src/main/scala/LuhnAlgorithm.scala

```diff
@@ -0,0 +1,45 @@
+object LuhnAlgorithm extends App{
+  //var digit = "5204408086566492"
+  def checkCard(digit:String)={
+    var digitStringList = digit.toString.toList.reverse
+    var extraWorkStuff = false
+    var addition = 0
+    def addNumbers(doExtraWork:Boolean,elem:Int)={
+      var number= elem
+      if (doExtraWork){
+        number = number*2
+        var numberList = number.toString.toList
+        numberList.foreach(num => addition+=num.toInt-48)
+
+
+
+      }
+      else{
+        addition += number
+      }
+    }
+
+    def checkOddOrEven(extraWork:Boolean):Boolean={
+      val doExtraWork = extraWork
+      if (doExtraWork){
```

Comment:
doExtraWork is immutable, to which you assign value of extraWork and then you use it to put into the if check and return it at the end, why not just use extraWork itself